### PR TITLE
Check mon connection scores post installation of cluster

### PR DESCRIPTION
**Pull request by vipin-m-s**
Automation of the issues - BZ-1892173, BZ-2174954, BZ-2151501, BZ-2315595

The steps are updated at - CEPH-83602911

Automation steps:-
1. Deploy a Ceph cluster
2. Validate correct number of mons in connection score,
correct number of peers for each mon, each mon present in quorum
and valid connection score ( score between 0 - 1 )
3. Change election strategy to disallow mode and perform
validations mentioned in step(2)
4. Change election strategy to connectivity mode and perform
validations mentioned in step(2)
5. Add new host as mon and perform validations
mentioned in step(2)
6. Remove mon added in step(5) in the cluster
and perform validation mentioned in step(2)
7. Restart monitor service using systemctl
and perform validations mentioned in step(2)
8. Stop monitor service using systemctl
and perform validations mentioned in step(2)
9. Start monitor service using systemctl
and validations mentioned in step(2)

Test case success logs :- http://magna002.ceph.redhat.com/ceph/ceph-qe-logs/vips/logs-vipin-automation-suite-test-3/

# Description

Please include Automation development guidelines. Source of Test case - New Feature/Regression Test/Close loop of customer BZs
<details>

<summary>click to expand checklist</summary>

- [ ] Create a test case in Polarion reviewed and approved.
- [ ] Create a design/automation approach doc. Optional for tests with similar tests already automated.
- [ ] Review the automation design
- [ ] Implement the test script and perform test runs
- [ ] Submit PR for code review and approve
- [ ] Update Polarion Test with Automation script details and update automation fields
- [ ] If automation is part of Close loop, update BZ flag qe-test_coverage “+” and link Polarion test
</details>

### CODE REVIEW

File: tests/rados/test_mon_connection_scores.py [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+                    return False
+
+                if (
+                    peer_score_detail["peer_score"] < 0
+                    and peer_score_detail["peer_score"] > 1
+                ):
+                    log.error("Invalid peer_score value in connection scores dump")
+                    return False
+
+    return True
```
Comment (pdhiran) [inline tests/rados/test_mon_connection_scores.py:246]: add new line at end

File: tests/rados/test_mon_connection_scores.py [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+            log.error("Test failed due to crash at the end of test")
+            return 1
+
+    log.info(
+        f"Successfully verified connection scores at different scenarios\
+         later reverted cluster to original state"
+    )
+    return 0
+
+def connection_score_checks(mon_nodes, rados_obj, num_mons, mon_election_obj) -> bool:
```
Comment (pdhiran) [inline tests/rados/test_mon_connection_scores.py:185]: Move the method to mon workflows

Comment (pdhiran) [inline tests/rados/test_mon_connection_scores.py:185]: Please add doc strings, args, examples of method call, return

Comment (vipin-m-s) [reply tests/rados/test_mon_connection_scores.py]: Moved connection_score_checks() to ceph/rados/monitor_workflows.py

Comment (vipin-m-s) [reply tests/rados/test_mon_connection_scores.py]: Added doc strings, args, examples of method call and return type

https://github.com/red-hat-storage/cephci/blob/95ed7c39e1c4c003d124590920d0fd0da31085c8/ceph/rados/monitor_workflows.py#L286-L313

**Pull request closed without merging**